postgres-appliance/bootstrap/clone_with_wale.py

```diff
@@ -70,20 +72,25 @@ def choose_backup(backup_list, recovery_target_time):
 
     match_timestamp = match = None
     for backup in backup_list:
-        last_modified = parse(backup['last_modified'])
+        last_modified = parse(backup['start_time' if os.getenv('USE_WALG_RESTORE') == 'true' else 'last_modified'])
```

Comment:
Hi Simon, should this be `finish_time` instead of `start_time` of the backup to compare with the `recovery_target_time`? Postgres requires that the recovery target "must be after the ending time of the base backup" (https://www.postgresql.org/docs/13/continuous-archiving.html).

Reply:
Thanks for bringing this up!

The rational behind picking `start_time` over `finish_time` is that one backup might have been started first but finished second. I'm actually not sure how likely parallel backups are, if they are possible at all. Please let me know if you know more on this matter!

Either way, after having read the documentation you linked to I think you are absolutely correct. It does not make sense to pick a base backup that has not yet finished before the "recovery target".

https://github.com/elastisys/spilo/pull/1/commits/272fbf171997f2c158a028938c7e4aabcfac3582